// backend/config/middlewares.js
module.exports = [
  'strapi::errors',
  {
    name: 'strapi::security',
    config: {
      contentSecurityPolicy: {
        useDefaults: true,
        directives: {
          'connect-src': ["'self'", 'http:', 'https:'],
          'img-src': [
            "'self'", 
            'data:', 
            'blob:', 
            'https://cyber-portfolio-72310aa69f55.herokuapp.com/', 
            'https://cyber-portfolio1.vercel.app',
            'https://*.amazonaws.com' 
          ],
          'media-src': [
            "'self'", 
            'data:', 
            'blob:', 
            'https://cyber-portfolio-72310aa69f55.herokuapp.com/',
            'https://cyber-portfolio1.vercel.app',
            'https://*.amazonaws.com' 
          ],
          upgradeInsecureRequests: null,
        },
      },
    },
  },
  {
    name: 'strapi::cors',
    config: {
      origin: ['https://cyber-portfolio1.vercel.app'],
      headers: ['*'],
    }
  },
  'strapi::poweredBy',
  'strapi::logger',
  'strapi::query',
  'strapi::body',
  'strapi::session',
  'strapi::favicon',
  'strapi::public',
];
